src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -1350,8 +1358,16 @@ bool PhaseIdealLoop::loop_predication_impl(IdealLoopTree* loop) {
   CountedLoopNode *cl = nullptr;
   if (head->is_valid_counted_loop(T_INT)) {
     cl = head->as_CountedLoop();
-    // do nothing for iteration-splitted loops
-    if (!cl->is_normal_loop()) return false;
+    if (!cl->is_normal_loop()) {
+      // Do nothing for iteration-splitted loops
+      return false;
+    }
+    loop->compute_trip_count(this);
+    if (cl->trip_count() == 1) {
+      // Not worth to hoist checks out of a loop that is only run for one iteration since the checks are only going to
+      // be executed once anyway.
+      return false;
+    }
     // Avoid RCE if Counted loop's test is '!='.
     BoolTest::mask bt = cl->loopexit()->test_trip();
     if (bt != BoolTest::lt && bt != BoolTest::gt)
```

Comment:
Since you touching this code can you add `{}` for this condition?

Reply:
Thanks Vladimir for your review! Yes, absolutely, I've pushed an update.